src/Data/Nat/Binary/Minus.agda

```diff
@@ -0,0 +1,239 @@
+Minus x y =  ℕᵇ × Tri (x < y) (x ≡ y) (x > y)
+
+minus :  ∀ x y → Minus x y                            -- extended subtraction
+minus zero     zero       =  (0ᵇ ,       tri-≈ refl)
+minus zero     (2[1+ _ ]) =  (0ᵇ ,       tri-< 0<even)
+minus zero     1+[2 _ ]   =  (0ᵇ ,       tri-< 0<odd)
+minus 2[1+ x ] zero       =  (2[1+ x ] , tri-> 0<even)
+minus 2[1+ x ] 2[1+ y ]   with minus x y
+... | (d , cmp) = (double d , even<even-mono-cmp cmp)
+minus 2[1+ x ] 1+[2 y ]  with minus x y
+... | (_ , tri< x<y _   _)   =  (0ᵇ  , tri-< (even<odd x<y))
+... | (d , tri≈ _   x≡y _)   =  (1+[2 d ] , tri-> (odd<even (inj₂ (sym x≡y))))
+... | (d , tri> _   _   x>y) =  (1+[2 d ] , tri-> (odd<even (inj₁ x>y)))
+minus 1+[2 x ] zero      =  (1+[2 x ] , tri-> 0<odd)
+minus 1+[2 x ] 2[1+ y ]  with minus x y
+... | (_ , tri< x<y _  _)   =  (0ᵇ , tri-< (odd<even (inj₁ x<y)))
+... | (_ , tri≈ _   eq _)   =  (0ᵇ , tri-< (odd<even (inj₂ eq)))
+... | (d , tri> _   _  x>y) =  (pred (double d) , tri-> (even<odd x>y))
+minus 1+[2 x ] 1+[2 y ]  with minus x y
+... | (d , cmp) =  (double d , odd<odd-mono-cmp cmp)
+
+infixl 6 _∸_
+
+_∸_ : Op₂ ℕᵇ              -- traditional subtraction
+_∸_ x =  proj₁ ∘ minus x
+
+0∸x≡0 :  ∀ x → 0ᵇ ∸ x ≡ 0ᵇ
+0∸x≡0 zero     =  refl
+0∸x≡0 2[1+ _ ] =  refl
+0∸x≡0 1+[2 _ ] =  refl
+
+x∸0≡x :  ∀ x → x ∸ 0ᵇ ≡ x
+x∸0≡x zero     =  refl
+x∸0≡x 2[1+ _ ] =  refl
+x∸0≡x 1+[2 _ ] =  refl
+
+even∸odd-for≥ :  ∀ {x y} → x ≥ y → 2[1+ x ] ∸ 1+[2 y ] ≡ 1+[2 (x ∸ y) ]
+even∸odd-for≥ {x} {y} x≥y  with minus x y
+... | (_ , tri< x<y _ _) =  contradiction x≥y (<⇒≱ x<y)
+... | (_ , tri≈ _   _ _) =  refl
+... | (_ , tri> _   _ _) =  refl
+
+odd∸even-for> :  ∀ {x y} → x > y → 1+[2 x ] ∸ 2[1+ y ] ≡ pred (double (x ∸ y))
+odd∸even-for> {x} {y} x>y  with minus x y
+... | (_ , tri< x<y _   _) =  contradiction x>y (<⇒≯ x<y)
+... | (_ , tri≈ _   x≡y _) =  contradiction x≡y (>⇒≢ x>y)
+... | (_ , tri> _   _   _) =  refl
+
+x≤y⇒x∸y≡0 :  ∀ {x y} → x ≤ y → x ∸ y ≡ 0ᵇ
+x≤y⇒x∸y≡0 {zero} {y} _                       =  0∸x≡0 y
+x≤y⇒x∸y≡0 (inj₁ (even<even x<y))             =  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 (inj₁ x<y))
+x≤y⇒x∸y≡0 {2[1+ x ]} {2[1+ .x ]} (inj₂ refl) =  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 {x} {x} (inj₂ refl))
+x≤y⇒x∸y≡0 {2[1+ x ]} {1+[2 y ]} (inj₁ (even<odd x<y))  with minus x y
+... | (0ᵇ , tri< _ _   _  ) =  refl
+... | (_ ,  tri≈ _ x≡y _  ) =  contradiction x≡y (<⇒≢ x<y)
+... | (_ ,  tri> _ _   x>y) =  contradiction x>y (<⇒≯ x<y)
+
+x≤y⇒x∸y≡0 {1+[2 x ]} {2[1+ y ]} (inj₁ (odd<even x≤y))  with minus x y
+... | (_ , tri< _ _ _  ) =  refl
+... | (_ , tri≈ _ _ _  ) =  refl
+... | (_ , tri> _ _ x>y) =  contradiction x>y (≤⇒≯ x≤y)
+
+x≤y⇒x∸y≡0 {1+[2 _ ]} {2[1+ _ ]} (inj₂ ())
+
+x≤y⇒x∸y≡0 {1+[2 _ ]} {1+[2 _ ]}  (inj₁ (odd<odd x<y)) =  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 (inj₁ x<y))
+x≤y⇒x∸y≡0 {1+[2 x ]} {1+[2 .x ]} (inj₂ refl)          =
+  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 {x} {x} (inj₂ refl))
+
+x∸x≡0 :  ∀ x → x ∸ x ≡ 0ᵇ
+x∸x≡0 x =  x≤y⇒x∸y≡0 (≤-refl {x})
+
+y<x⇒x∸y≢0 :  ∀ {x y} → y < x → x ∸ y ≢ 0ᵇ
```

Comment:
The convention is that the variables should be allocated names in the order that they appear, i.e. this should be either `x<y⇒y∸x≢0` or `x>y⇒x∸y≢0` rather than `y<x⇒x∸y≢0`

Reply:
Changed to ``x>y⇒x∸y≢0``. The same done to a certain another function.